teaser/data/output/besmod_output.py

```diff
@@ -0,0 +1,432 @@
+    THydSup_nominal : float or dict, optional
+        Nominal supply temperature(s) for the hydraulic system. Required for
+        certain examples (e.g., HeatPumpMonoenergetic, GasBoilerBuildingOnly).
+        See docstring of teaser.data.output.besmod_output.convert_input() for further information.
+    QBuiOld_flow_design : dict, optional
+        For partially retrofitted systems specify the old nominal heat flow
+        of the Buildings in a dictionary with the building names as keys.
+        By default, only the radiator transfer system is not retrofitted in BESMod.
+    THydSupOld_design : float or dict, optional
+        Design supply temperatures for old not retrofitted hydraulic systems.
+    custom_examples: dict, optional
+        Specify custom examples with a dictionary containing the example name as the key and
+        the path to the corresponding custom mako template as the value.
+    custom_script: dict, optional
+        Specify custom .mos scripts for the existing and custom examples with a dictionary
+        containing the example name as the key and the path to the corresponding custom mako template as the value.
+
+    Raises
+    ------
+    ValueError
+        If given example is not supported.
+    ValueError
+        If `THydSup_nominal` is not provided for examples requiring it.
+    AssertionError
+        If the used library for calculations is not AixLib.
+    NotImplementedError
+        If a building uses a thermal zone model other than the four-element model.
+
+    Notes
+    -----
+    The function uses Mako templates for generating Modelica models.
+    """
+
+    if prj.used_library_calc != "AixLib":
+        raise AttributeError("BESMod export is only implemented for AixLib calculation.")
+    if examples is None:
+        examples = []
+    if not isinstance(examples, list):
+        examples = [examples]
+    supported_examples = ["TEASERHeatLoadCalculation",
+                          "HeatPumpMonoenergetic",
+                          "GasBoilerBuildingOnly"]
+    for exp in examples:
+        if exp not in supported_examples:
+            raise ValueError(f"Example {exp} is not supported. "
+                             f"Supported examples are {supported_examples}.")
+
+    if THydSup_nominal is None and any(
+            example in examples for example in ["HeatPumpMonoenergetic", "GasBoilerBuildingOnly"]):
+        raise ValueError("Examples 'HeatPumpMonoenergetic' and 'GasBoilerBuildingOnly' "
+                         "require the `THydSup_nominal` parameter.")
+
+    t_hyd_sup_nominal_bldg = convert_input(THydSup_nominal, buildings)
+    t_hyd_sup_old_design_bldg = convert_input(THydSupOld_design, buildings) if THydSupOld_design else \
+        {bldg.name: "systemParameters.THydSup_nominal" for bldg in buildings}
+
+    if QBuiOld_flow_design is None:
+        QBuiOld_flow_design = {bldg.name: "systemParameters.QBui_flow_nominal" for bldg in buildings}
+    else:
```

Comment:
Maybe apply black code formatting, the syntax looks at least weird on a small screen.